sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -68,23 +68,25 @@ class AstBuilder extends DataTypeAstBuilder with SQLConfHelper
 
   protected def withIdentClause(
```

Comment:
can we create an overload of this method that takes no `otherPlans`? Then we can reduce a lot of code changes of this PR.